orm/helpers.go

```diff
@@ -0,0 +1,70 @@
+package orm
+
+import (
+	"reflect"
+
+	"github.com/iov-one/weave/errors"
+)
+
+type limitedIterator struct {
+	// n is count of remaining elements
+	n int
+	// iterator is the underlying iterator
+	iter SerialModelIterator
+}
+
+var _ SerialModelIterator = (*limitedIterator)(nil)
+
+func (l *limitedIterator) LoadNext(dest SerialModel) error {
+	if l.n > 0 {
+		l.n--
+		return l.iter.LoadNext(dest)
+	}
+	return errors.Wrap(errors.ErrIteratorDone, "iterator limit reached")
+}
+
+func (l *limitedIterator) Release() {
+	l.iter.Release()
+}
+
+func WithLimit(iter SerialModelIterator, limit int) (SerialModelIterator, error) {
```

Comment:
I think this function could be improved.

Rename `WithLimit` to `LimitIterator`, to match [`io.LimitReader`](https://golang.org/pkg/io/#LimitReader). I think `Limit` would also be a good name.

Removing the error would make much nicer API. It would be ```Limit(SerialModelIterator, int) SerialModelIterator``` and the caller does not have to test error. Giving a limit value less than 1 makes little sense, but it is not invalid. Creating a limit of 0 or less will produce a valid iterator that always returns `ErrIteratorDone`.